### PR TITLE
[ERSSUP-64033]-[]-[Application-restricted OAS Document Update for A02…

### PR DESCRIPTION
**Pull request by RajaGovindharaj16**
…4, A025 and A043 Endpoints]-[RG]

## Summary
* Routine Change
* :exclamation: Breaking Change
* :robot: Operational or Infrastructure Change
* :sparkles: New Feature
* :warning: Potential issues that might be caused by this change

Add any other relevant notes or explanations here. **Remove this line if you have nothing to add.**

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.

* [ ] I have reviewed the changes in this PR and they fill all or part of the acceptance criteria of the ticket, and the code is in a mergeable state.
* [ ] If there were infrastructure, operational, or build changes, I have made sure there is sufficient evidence that the changes will work.
* [ ] I have ensured the changelog has been updated by the submitter, if necessary.

### CODE REVIEW

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml [hunk 1]
```diff
@@ -9,22 +9,29 @@ description: |
   The conversation can be retrieved via [[HYPERLINK_A025]].
 
   ## Supported security patterns
+  - Application-restricted access, unattended access
```
Comment (jamiegreen4) [inline specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:12]: Has it been queried why this mentions 'unattended access' and the others don't?

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml [hunk 2]
```diff
@@ -36,6 +43,16 @@ description: |
...
+  These may be identified using [[HYPERLINK_A005]] and will be returned if the user is authourised to view this.
+
+  If the converted advice and guidance request is referred to a service outside of your organisation, 
+  the subsequent referral information may not be accessible due to existing data-access rules. However, 
+  the advice and guidance summary ([[HYPERLINK_A024]]), attachments ([[HYPERLINK_A006]]), advice and guidance conversation ([[HYPERLINK_A025]]) and 
+  clinical information ([[HYPERLINK_A043]]) which took place before such a conversion still forms part of the clinical information and can still be retrieved.
+
   ## Related endpoints
 
   After using this endpoint, use [[HYPERLINK_A025]] to retrieve the Advice and Guidance conversation itself.
```
Comment (jamiegreen4) [inline specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:58]: I'm assuming the info about A043 in the Related endpoints is still required even though it's not mentioned on ERSSUP-63512?

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml [hunk 3]
```diff
@@ -14,6 +15,10 @@ description: |
 
   Note: e-RS only supports the retrieval of the most recent version of `CommunicationRequest`. You cannot retrieve earlier versions.
 
+  ## Important note
+ 
+  The Important notes for this endpoint are the same as for endpoint [[HYPERLINK_A024]].
+
   ## Use case
 
   See [[HYPERLINK_A005]].
```
Comment (jamiegreen4) [inline specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml:24]: Should headers NHSD-End-User-Organisation-ODS, NHSD-eRS-Business-Function and NHSD-eRS-On-Behalf-Of-User-ID mention 'not allowed for application-restricted access' and 'Required for user-restricted access'? This goes for all 4 endpoints

File: specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml [hunk 4]
```diff
@@ -35,6 +42,16 @@ description: |
...
+  These may be retrieved from [[HYPERLINK_A024]].
+
+  If the converted advice and guidance request is referred to a service outside of your organisation, 
+  the subsequent referral information may not be accessible due to existing data-access rules. 
+  However, the advice and guidance summary ([[HYPERLINK_A024]]), attachments ([[HYPERLINK_A006]]), 
+  advice and guidance conversation ([[HYPERLINK_A025]]) and 
+  clinical information ([[HYPERLINK_A043]]) which took place before such a conversion still forms part of the clinical information and can still be retrieved.
+  The conversation also gets included in the form of a PDF file returned by [[HYPERLINK_A007]]
 
   ## Related endpoints
```
Comment (jamiegreen4) [inline specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml:57]: Should the sandbox work for A025? Also On ERSSUP-63516 it shows GET/CommunicationRequest/{ubrn} but the 'Try this API' link doesn't include the /{ubrn}

File: specification/components/stu3/schemas/endpoints/a043-retrieve-advice-and-guidance-overview-pdf.yaml [hunk 5]
```diff
@@ -7,32 +7,39 @@ description: |
...
+  As such, it is important that all clinical information is obtained across all the related UBRNs in the pathway. 
+  These can be identified using [[HYPERLINK_A005]].
 
-  The advice and guidance conversation that took place before such a conversion still forms part of the clinical information. 
-  This conversation *does* get included in the PDF file returned by [[HYPERLINK_A007]].
+  If the converted advice and guidance request is referred to a service outside of your organisation, 
+  the subsequent referral information may not be accessible due to legitimate relationship rules. 
+  However, the advice and guidance summary ([[HYPERLINK_A024]]), 
+  attachments ([[HYPERLINK_A006]]), advice and guidance conversation ([[HYPERLINK_A025]]) and 
+  clinical information ([[HYPERLINK_A043]]) which took place before such aconversion still forms part of the clinical information and can still be retrieved.
```
Comment (jamiegreen4) [inline specification/components/stu3/schemas/endpoints/a043-retrieve-advice-and-guidance-overview-pdf.yaml:38]: type missing space in aconversion

File: specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml (hunk 4 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml:57]: Yes it should work. The actual endpoint url is "/STU3/Communication". I think it was copied from A024 endpoint to here https://nhsd-jira.digital.nhs.uk/browse/ERSSUP-63516.  I will raise this to BA

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml (hunk 1 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:12]: Raised to BA and BA updated all 4.

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml (hunk 3 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml:24]: Raised with BA and they need to check with NHSD

File: specification/components/stu3/schemas/endpoints/a043-retrieve-advice-and-guidance-overview-pdf.yaml (hunk 5 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a043-retrieve-advice-and-guidance-overview-pdf.yaml:38]: Fixed.

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml (hunk 1 above)
Comment (jamiegreen4) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:12]: Does that mean the references on the existing endpoints should also be updated to be consistent? NHSD were very keen on consistency during the first phase of app restricted OAS changes

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml (hunk 2 above)
Comment (jamiegreen4) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:58]: I wasn't sure it should be removed, I was more thinking it was missing from the story?

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml (hunk 3 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance-versioned.yaml:24]: NHSD approved the changes. So we are going to include the OAS documentation on the headers

File: specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml (hunk 4 above)
Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a025-retrieve-advice-and-guidance-conversation.yaml:57]: BA updated the Story

File: specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml (hunk 1 above)
Comment (jamiegreen4) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:12]: I can see that in addition to the A&G endpoints, A004 has also bee updated with 'unattended access'. Should the other earlier app restricted endpoints also mention this?

Comment (RajaGovindharaj16) [reply specification/components/stu3/schemas/endpoints/a024-retrieve-advice-and-guidance.yaml:12]: I had a TL review yesterday with Adam and Jack and also spoke to Justine. There are few changes introduced part of this change including "unattended access" and some text ordering on pre-requisites are to be done through a wider piece of work at the whole OAS documentation change.

**Comment by jamiegreen4** [comment]
Changes look fine to me

**Review by jamiegreen4 — APPROVED** [review]

**Pull request merged**